tests/consensus/test_soft_voided.py

```diff
@@ -43,8 +43,12 @@ def _run_test(self, simulator, soft_voided_tx_ids):
         gen_tx2 = simulator.create_tx_generator(manager2, rate=10 / 60., hashpower=1e6, ignore_no_funds=True)
         gen_tx2.start()
 
-        while not gen_tx2.latest_transactions:
+        for _ in range(100):
+            if gen_tx2.latest_transactions:
+                break
             simulator.run(600)
+        else:
+            self.fail('took too long to generate a transaction')
```

Comment:
Why don't we move this piece to `gen_tx2.run_until_find_a_transaction()`? It would prevent duplicated code on test_soft_voided, test_soft_voided3, and test_soft_voided4 files.

Reply:
I added a `gen_tx2.yield_until_find_a_transaction(max_steps=100)` which includes most of the code, it leaves out the `simulator.run(600)` is left to the user of `yield_until_find_a_transaction`, since it varies a little bit on the other cases.

Reply:
Well done. Thanks!